src/lib/Inference.hs

```diff
@@ -1982,8 +2020,14 @@ instance Unifiable Atom where
          unless (c == c') empty >> unifyFoldable xs xs'
        (TC con, TC con') -> unifyFoldable con con'
        (Eff eff, Eff eff') -> unify eff eff'
+       (DictTy d, DictTy d') -> unify d d'
        _ -> empty
 
+instance Unifiable DictType where
+  unifyZonked (DictType _ c params) (DictType _ c' params')
+    | c == c'   = unifyFoldable params params'
+    | otherwise = empty
```

Comment:
🏌️‍♂️ This can also be spelled as `guard (c == c') *> unifyFoldable params params'`